### PR TITLE
{AppService} `az webapp`: Add parameter `--acr-identity` for `az webapp create` to allow users choose user assigned identity for acr image pull and add parameters `--acr-use-identity` and `--acr-identity` for `az webapp config set` to update acr image pull related configs

### PR DESCRIPTION
**Pull request by mstxsun**
**Related command**

`az webapp create`  and `az webapp config set`

**Description**

1. Add parameter `--acr-identity` for `az webapp create` to allow users choose user assigned identity they want to use for ACR image pull.
2. Add parameters `--acr-use-identity` and `--acr-identity` for `az webapp config set` to allow users update ACR image pull related configs.

**Testing Guide**

1. Ran commands `az webapp create` and `az webapp config set` locally
2. Ran tests test_webapp_hyperv_acr_use_identity and test_webapp_linux_acr_use_identity

**History Notes**

[AppService] `az webapp create`: Add parameter `--acr-identity` to allow users choose user assigned identity they want to use for ACR image pull
[AppService] `az webapp config set`: Add parameter `--acr-use-identity` and `--acr-identity` to allow users update ACR image pull related configs

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by yanzhudd** [comment]
please follow this guideline of developing managed identity: https://github.com/Azure/azure-cli/blob/20ffb2d35d59a104641d75aeb76b6f31885e26aa/doc/managed_identity_command_guideline.md

**Comment by mstxsun** [comment]
> please follow this guideline of developing managed identity: https://github.com/Azure/azure-cli/blob/20ffb2d35d59a104641d75aeb76b6f31885e26aa/doc/managed_identity_command_guideline.md

Hi @yanzhudd, this PR isn't used to add or operate managed identity on existing resource. We currently already have an old command --assign-identity which is used to create identity subgroup and attach managed identities to newly created webapp. For my change, it is used to let customer decide which managed identity they already assigned to their newly created webapp by using --assign-identity, they are going to use for authentication when pulling image from ACR(Azure container registry) I believe this is a scenario which is different to what you guys mentioned in that document. Changes on az webapp config set is similar, it is not used to manage managed identities directly.

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py [hunk 1]
```diff
@@ -1671,6 +1671,9 @@
     text: >
         AppServicePlanID=$(az appservice plan show -n SharedAppServicePlan -g MyASPRG --query "id" --out tsv)
         az webapp create -g MyResourceGroup -p "$AppServicePlanID" -n MyUniqueAppName
+  - name: create a container webapp with an image pulled from a private Azure Container Registry using a User Assigned Managed Identity
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:1674]:
```suggestion
  - name: Create a container webapp with an image pulled from a private Azure Container Registry using a User Assigned Managed Identity
```

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 2]
```diff
@@ -356,6 +358,9 @@ def load_arguments(self, _):
                 c.ignore('reserved_instance_count')
                 c.argument('runtime', help="Canonicalized web runtime in the format of Framework:Version, e.g. \"PHP:7.2\"."
                                            "Use `az webapp list-runtimes` for available list")
+                c.argument('acr_use_identity', arg_type=get_three_state_flag(return_label=True), help="Enable or disable pull image from acr use managed identity")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:361]:
```suggestion
                c.argument('enable_acr_identity', arg_type=get_three_state_flag(return_label=True), help="Enable or disable pull image from acr use managed identity")
```

**Comment by zhoxing-ms** [comment]
@mstxsun Please note that Azure CLI will have a **code freeze** on **07/30/2024 10:00 UTC** for the upcoming release. If you want to catch this release train, please address all the comments by Tomorrow, otherwise it has to be postponed to next sprint (09-03).

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py (hunk 2 above)
Comment (mstxsun) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:361]: Hi @zhoxing-ms, we already have --acr-use-identity in az webapp create, so we are using this to keep parameters in different commands consistent

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:361]: Got it, thanks~

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**